src/formState.test.tsx

```diff
@@ -1375,6 +1376,103 @@ describe("formState", () => {
     expect(formState.books.rows[1].isNewEntity).toBeTruthy();
     expect(formState.books.isNewEntity).toBeFalsy();
   });
+
+  it("keeps local changed values when a query refreshes", async () => {
+    // Given a component
+    function TestComponent() {
+      type FormValue = AuthorInput;
+      const config: ObjectConfig<FormValue> = authorWithAddressAndBooksConfig;
+      // And we have two sets of data
+      const data1 = {
+        id: "a:1",
+        firstName: "f1",
+        lastName: "l1",
+        address: { id: "address:1", street: "s1", city: "c1" },
+        books: [
+          { id: "b:1", title: "a1" },
+          { id: "b:2", title: "b1" },
+        ],
+      };
+      const data2 = {
+        id: "a:1",
+        firstName: "f2",
+        lastName: "l2",
+        address: { id: "address:1", street: "s2", city: "c2" },
+        books: [
+          { id: "b:1", title: "a2" },
+          { id: "b:2", title: "b2" },
+          { id: "b:3", title: "b3" },
+        ],
+      };
+      // And we start out with data1
+      const [data, setData] = useState<FormValue>(data1);
+      const form = useFormState({ config, init: { input: data, map: (d) => d } });
+      function makeLocalChanges() {
+        form.firstName.value = "local";
+        form.address.street.value = "local";
+        form.books.rows[0].title.value = "local";
```

Comment:
**thought (non-blocking):** Would it be better to use our setters vs set the value directly, just to do it the way we expect apps to interact with the objects? i.e. `form.firstName.set("local")`. Appears we get the same result regardless.

Reply:
Ah yeah, this is the same thing as calling the `set` methods...to a certain extent, we probably shouldn't have both, i.e. two ways of doing the same thing, but also it doesn't hurt... :shrug: I think it's fine.